src/main/java/org/reactome/release/dataexport/configuration/ConfigurationManager.java

```diff
@@ -0,0 +1,274 @@
+	 * more of the needed property values for the project).  If the file is (re)created, the file permissions will be
+	 * set to 660 to allow read and write access to the user and group only.
+	 *
+	 * @param overwriteExistingFile <code>true</code> if the file should be (re)created regardless if it already exists
+	 * or is valid;<code>false</code> otherwise
+	 * @return <code>true</code> if the configuration file is (re)created because the file does not exist, is not
+	 * valid, or was requested to be overwritten;<code>false</code> otherwise (i.e the configuration file already
+	 * exists, is valid, and was not requested to be overwritten)
+	 * @throws IOException  Thrown if unable to read an existing configuration file or if unable to write a new
+	 * configuration file if one needs to be recreated (this includes an exception from deleting an existing file,
+	 * writing a new file, or setting its file system permissions appropriately).
+	 */
+	public boolean createConfigurationFile(boolean overwriteExistingFile) throws IOException {
+		if (!overwriteExistingFile && configurationFileExistsAndIsValid()) {
+			return false;
+		}
+
+		initializeConfigurationEntryCreator();
+
+		addNeo4jDatabaseConfigurationEntries();
+		addReactomeSpecificConfigurationEntries();
+		addEuropePMCFTPServerConfigurationEntries();
+		addNCBIFTPServerConfigurationEntries();
+
+		if (!configurationFileExists()) {
+			System.out.println(
+				"The configuration file " + getConfigFileName() + " does not exist.  "
+					+ "Please provide the following property values.");
+		} else if (!configurationFileIsValid()) {
+			System.out.println(
+				"The configuration file " + getConfigFileName() + " does not have all the required property "
+					+ "values.  Please provide the following property values and the configuration file will be "
+					+ "recreated"
+			);
+		}
+
+		writeConfigurationFile();
+
+		return true;
+	}
+
+	/**
+	 * Stop git tracking on original/sample configuration file to prevent committing and pushing if any sensitive
+	 * information is mistakenly added.  If the system command to stop git tracking the configuration file returns a
+	 * non-zero exit value, a warning indicating this will be logged.
+	 *
+	 * @throws IOException Thrown if the system command to stop git tracking produces an I/O Error
+	 */
+	public void stopGitTrackingOriginalSampleConfigurationFile() throws IOException {
+		String originalConfigurationFilePath = getPathToOriginalSampleConfigurationFile();
+		String stopGitTrackingCommand = "git update-index --assume-unchanged " + originalConfigurationFilePath;
+
+		Process stopGitTrackingProcess = Runtime.getRuntime().exec(stopGitTrackingCommand);
+		if (stopGitTrackingProcess.exitValue() != 0) {
+			logger.warn(
+				"Exit value for command to stop git tracking for file " + originalConfigurationFilePath + " was " +
+				stopGitTrackingProcess.exitValue()
+			);
+		};
+	}
```

Comment:
This method should not be necessary - it should not exist.

Reply:
Stopping the tracking of files is probably something that should be done in the script that calls this code - not in the code itself.
This method makes the very specific assumption that the code will be compiled and run from a directory that is tracked by git. What will happen if the code is run in a directory that is _not_ tracked by git?

Reply:
Fair point - I can move it back to the shell script.

Reply:
Done: https://github.com/reactome/release-data-exporter/pull/4/commits/5063fc70c2e1d1b2205731b96706981aec38bab0